### PR TITLE
sc-4575 Add notebook UI

### PR DESCRIPTION
**Pull request by pdeziel**
This adds a simple notebook UI using tkinter which can be attached to a `DistributedNotebook`. If attached, the UI will send updates to the `DistributedNotebook` when the user adds/removes cells or edits the text in a cell.

Also in this PR is a refactor of the `DistributedNotebook` and `Cell` classes. They now are just child classes of `Sequence` and a notebook cell just contains a sequence of characters rather than a sequence of lines. This simplifies the implementation a lot and makes it work better with the UI.

### CODE REVIEW

File: notebook/cell.py [hunk 1]
```diff
@@ -1,74 +1,49 @@
-import uuid
+import difflib
```
Comment (rebeccabilbro) [inline notebook/cell.py:1]: what a useful find!

File: tests/test_cell.py [hunk 2]
```diff
@@ -16,79 +16,47 @@ def test_single_cell(self):
...
+        a.append_text("\nthis is the second line")
+        assert a.get_text() == "this is the first line\nthis is the second line"
+
+        a.remove_many(0, 5)
+        assert a.get_text() == "is the first line\nthis is the second line"
+
+        a.insert_text(7, "edited ")
+        assert a.get_text() == "is the edited first line\nthis is the second line"
+
+    @pytest.mark.parametrize(
```
Comment (rebeccabilbro) [inline tests/test_cell.py:29]: nice use of parametrize!

**Review by rebeccabilbro — APPROVED** [review]
This looks really good @pdeziel ! Thank you for walking me through everything yesterday; the UI component is pretty cool 😎

File: notebook/cell.py (hunk 1 above)
Comment (pdeziel) [reply notebook/cell.py:1]: standard library too!

**Pull request merged**